diff --git a/set-wallpaper-contract/CMakeLists.txt b/set-wallpaper-contract/CMakeLists.txt
index 849f118..a5e1c22 100644
--- a/set-wallpaper-contract/CMakeLists.txt
+++ b/set-wallpaper-contract/CMakeLists.txt
@@ -1,14 +1,14 @@
 set (EXEC_NAME "set-wallpaper")
 set (SOURCES "set-wallpaper.vala")
 set (SWEXECDIR "${CMAKE_INSTALL_FULL_LIBEXECDIR}/switchboard-plug-pantheon-shell")
-set (DEPENDENCIES granite posix)
+set (DEPENDENCIES granite)
 
 configure_file (set-wallpaper.contract.in ${CMAKE_CURRENT_BINARY_DIR}/set-wallpaper.contract)
 
 find_package (PkgConfig)
 pkg_check_modules (DEPS REQUIRED ${DEPENDENCIES})
 
-vala_precompile (WALLPAPER_CONTRACT_VALA_C ${EXEC_NAME} ${SOURCES} PACKAGES ${DEPENDENCIES})
+vala_precompile (WALLPAPER_CONTRACT_VALA_C ${EXEC_NAME} ${SOURCES} PACKAGES ${DEPENDENCIES} posix)
 add_executable (${EXEC_NAME} ${WALLPAPER_CONTRACT_VALA_C})
 
 link_directories (${DEPS_LIBRARY_DIRS})
